velox/functions/prestosql/Repeat.cpp

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/expression/DecodedArgs.h"
+#include "velox/expression/VectorFunction.h"
+
+namespace facebook::velox::functions {
+namespace {
+
+// See documentation at https://prestodb.io/docs/current/functions/array.html
+class RepeatFunction : public exec::VectorFunction {
+ public:
+  static constexpr int32_t kMaxResultEntries = 10'000;
+
+  bool isDefaultNullBehavior() const override {
+    // repeat(null, n) returns an array of n nulls.
+    return false;
+  }
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& outputType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    const auto numRows = rows.size();
+    int32_t constantCount = 0;
```

Comment:
Thank you for revising the implementation. It looks good now.

I wonder if the code will be easier to read if processing of constant 'count' is fully separated from non-constant 'count'. For example, we don't need decodedArgs, nulls, rawNulls if 'count' is constant.

```
ArrayVectorPtr localResult;
if (args[1]->isConstantEncoding()) {
   localResult = applyConstant(...);
} else {
   ...
   localResult = ...
}

context.moveOrCopyResult(localResult, rows, result);
```

Reply:
I do agree this makes the code more readable. I tried this structure in the early revisions. It was pointed out that there were some repeated code and eventually I put them back together so they can share the common parts like memory allocation. I'll take a second look how to structure it. Thanks!